deploy-service/common/src/main/java/com/pinterest/deployservice/scm/SourceControlManagerProxy.java

```diff
@@ -31,8 +34,20 @@ private SourceControlManager getSourceControlManager(String scmType) throws Exce
         return manager;
     }
 
-    public String getType() {
-        return DEFAULT_TYPE;
+    public List<String> getSCMs() throws Exception {
+        validSCMs = new ArrayList<String> (this.managers.keySet());
```

Comment:
you can move the init of validSCM to the constructor above so this get will be just returning a variable.

Reply:
Makes sense. I will update accordingly.